mmv1/templates/terraform/constants/integration_connectors_connection.go.erb

```diff
@@ -0,0 +1,20 @@
+<% unless compiler == "terraformgoogleconversion-codegen" -%>
+// waitforConnectionReady waits for an connecion to leave the
+// "CREATING" state, to indicate that it's ready.
+func waitforConnectionReady(d *schema.ResourceData, config *transport_tpg.Config, timeout time.Duration) error {
+	return resource.Retry(timeout, func() *resource.RetryError {
+	if err := resourceIntegrationConnectorsConnectionRead(d, config); err != nil {
```

Comment:
why do we need to read the resource first?

Reply:
To get the resource object so that we can read the status of the connector resource.

Reply:
This will be updated to a stable state(not creating or updating) eventually and then we can allow other operations on this resource.

Reply:
my bad, read it wrong, I thought it was pre-create file somehow. Sorry!